### PR TITLE
Fix shutdown of partially started server

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Previously when a feature returned an error during the startup phase we would shut down all started features, but also only unprepare those started features. Features that were prepared but not yet started would not be unprepared.

In the ClusterFeature we would create and start the AgencyCache thread during prepare. In a test scenario we cancel the start before we call ClusterFeature::start, so previously this feature would neither be stopped or unprepared. Thus the AgencyCache thread would keep running and potentially access data and features that are getting cleaned up during the shutdown.

- [x] :hankey: Bugfix

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.12.0: *(Please link PR)*
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

### CODE REVIEW

File: arangod/Cluster/ClusterFeature.cpp [hunk 1]
```diff
@@ -680,18 +680,6 @@ void ClusterFeature::prepare() {
   }
 
   // This must remain here for proper function after hot restores
```
Comment (jsteemann) [inline arangod/Cluster/ClusterFeature.cpp:682]: The original code comment ("this must remain here...") is now obsolete. I am wondering if the comment was actually correct, now that you moved the agency cache startup to somewhere else.

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by kvahed — APPROVED** [review]
LGTM

File: arangod/Cluster/ClusterFeature.cpp (hunk 1 above)
Comment (mpoeter) [reply arangod/Cluster/ClusterFeature.cpp]: Discussed with @kvahed - seems to be outdated - removed!

**Pull request merged**